ui/src/app/utils/research-purpose.tsx

```diff
@@ -58,6 +62,15 @@ export function getSelectedPopulations(researchPurpose: ResearchPurpose) {
     .subCategory.filter(item => researchPurpose.populationDetails.includes(item.shortName)).length > 0));
   categories.filter(category => category.subCategory = category.subCategory
     .filter(subCategoryItem => researchPurpose.populationDetails.includes(subCategoryItem.shortName)));
```

Comment:
Again - existing issue, but I am very confused by the outer filter here.

```suggestion
  categories.forEach(category => {
    category.subCategory = category.subCategory.filter(({shortName}) => researchPurpose.populationDetails.includes(shortName));
  });
```